File: account/src/main/java/serp/project/account/core/domain/dto/request/CreateRoleDto.java
package serp.project.account.core.domain.dto.request;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class CreateRoleDto {
    @NotBlank
    private String name;
    private String description;
    @Builder.Default
    private Boolean isRealmRole = false;
    private String keycloakClientId;
    private Integer priority;
    @NotBlank
    private String scope;
    private Long scopeId;
    private Long moduleId;
    private Long organizationId;
    private Long departmentId;
    private Long parentRoleId;
    @NotBlank
    private String roleType;
    private Boolean isDefault;

    private List<Long> permissionIds;
}

File: api_gateway/src/core/domain/dto/request/account/create_role_dto.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package request

type CreateRoleDto struct {
	Name             string  `json:"name"`
	Description      *string `json:"description"`
	IsRealmRole      *bool   `json:"isRealmRole"`
	KeycloakClientId *string `json:"keycloakClientId"`
	Priority         *int    `json:"priority"`
	Scope            *string `json:"scope"`
	ScopeId          *int64  `json:"scopeId"`
	ModuleId         *int64  `json:"moduleId"`
	OrganizationId   *int64  `json:"organizationId"`
	DepartmentId     *int64  `json:"departmentId"`
	ParentRoleId     *int64  `json:"parentRoleId"`
	RoleType         *string `json:"roleType"`
	IsDefault        *bool   `json:"isDefault"`

	PermissionIds []int64 `json:"permissionIds"`
}
